transaction/summary/src/verifier.rs

```diff
@@ -117,7 +46,17 @@ pub fn verify_tx_summary(
 /// implementation details of the mc-crypto-digestible scheme.
 /// If TxSummary digestible annotations are changed then this object's
 /// implementation needs to change also.
-pub struct TxSummaryStreamingVerifier {
+pub struct TxSummaryStreamingVerifier<const RECORDS: usize = MAX_RECORDS> {
```

Comment:
`TxSummaryStreamingVerifier` only ended up being used internally for `verify_tx_summary` (because separating the verifier and report helps with stack use), could make this non-public

Reply:
This is a bit surprising to me -- when I wrote the code, `verify_tx_summary` was only there for ease of testing

I need to read the PR more carefully still

Reply:
I don't understand, does it mean that you only use `verify_tx_summary` in the hardware wallet?

The `verify_tx_summary` requires all the `TxSummary` and `TxSummaryUnblindingData` to be available, so does that mean we don't stream it?

Reply:
I think even if you aren't using it, it still makes sense to make the `TxSummaryStreamingVerifier` public in case someone else wants to actually take advantage of the streaming part.

Reply:
yeah the naming is confusing atm (sorry)... `TxSummaryStreamingVerifierCtx` is the streaming verifier context, used by the hardware wallet, `TxSummaryStreamingVerifier` just combines that with the report for easier handling, which is then called in `verify_tx_summary`.

i think probably i should remove the `TxSummaryStreamingVerifier` impl because if you need that function it's already represented by the `TxSummaryStreamingVerifierCtx` object and it's not much more complex to call this more directly from `verify_tx_summary`